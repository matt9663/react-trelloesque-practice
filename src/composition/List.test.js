import React from 'react';
import ReactDOM from 'react-dom';
import List from './List.js';
import renderer from 'react-test-renderer';


describe('List component', () => {
    it('renders without crashiing', () =>{
        const cards = [{
            title: 'test1',
            key: 'a'
        },
        {
            title: 'test2',
            key: 'b'
        }
    ];
        const div = document.createElement('div');
        ReactDOM.render(<List header="tester"
            cards={cards}/>, div);
        ReactDOM.unmountComponentAtNode(div);
    });
    it('renders the UI as expected', () => {
        const cards = [{
            title: 'test1',
            key: 'a'
        },
        {
            title: 'test2',
            key: 'b'
        }
    ];
        const tree = renderer
            .create(<List header="tester" cards={cards}/>)
            .toJSON();
    expect(tree).toMatchSnapshot();   
    });
});