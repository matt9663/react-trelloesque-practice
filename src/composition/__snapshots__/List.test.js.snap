// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List component renders the UI as expected 1`] = `
<section
  className="List"
>
  <header
    className="List-header"
  >
    <h2>
      tester
    </h2>
    <div
      className="List-cards"
    >
      <div
        className="Card"
      >
        <button
          type="button"
        >
          delete
        </button>
        <h3>
          test1
        </h3>
        <p />
      </div>
      <div
        className="Card"
      >
        <button
          type="button"
        >
          delete
        </button>
        <h3>
          test2
        </h3>
        <p />
      </div>
      <button
        className="List-add-button"
        type="button"
      >
        + Add Random Card
      </button>
    </div>
  </header>
</section>
`;
